### ISSUE TITLE
Improve Kimi K2 KL regularization

### ISSUE THREAD

**Issue by bbartoldson** [PrimeIntellect-ai/prime-rl#1390]
The current KL regularization approach does not provide off-policy robustness, as shown in the ablation study below.

<img width="1162" height="374" alt="Image" src="https://github.com/user-attachments/assets/e3ed5ec1-d5e4-4d40-8dd2-f30834844bee" />

This can be fixed by changing the code on [this line](https://github.com/PrimeIntellect-ai/prime-rl/blob/main/src/prime_rl/trainer/rl/loss.py#L102):

```
loss = loss + loss_config.kl_tau * (log_importance_ratio[kl_mask]).sum()
```

to be

```
loss = loss + loss_config.kl_tau * (log_importance_ratio[kl_mask].sum().detach() * importance_ratio[kl_mask]).sum()
```

Note that the latter keeps the log importance ratio (the KL proxy) in the gradient and multiplies this by the importance ratio (multiplying the KL term by the importance ratio is also done in [DeepSeek-V3.2](https://arxiv.org/pdf/2512.02556v1)).

The gradient of the former is incorrect relative to the Kimi K2 paper -- e.g. it lacks a denominator because `log_importance_ratio = log_trainer_probs - log_inference_probs`, and the gradient of the latter term is 0. The correct Kimi K2 gradient includes the (detached) log importance ratio as an advantage-like term, as shown below ([source](https://arxiv.org/abs/2503.18929)).

<img width="931" height="589" alt="Image" src="https://github.com/user-attachments/assets/36b38077-a13b-4ebc-8885-33571acbb3a5" />

With the above fix (and CISPO-style IS clipping instead of IcePop masking), the K2 KL approach is highly performant on off-policy data, similar to TBA, using $$\tau=0.005$$.

<img width="934" height="250" alt="Image" src="https://github.com/user-attachments/assets/56ec7592-c88a-4d19-afe1-6a8a2b2d62db" />

**Comment by bbartoldson** [comment]
This post-fix ablation study might also be interesting:

<img width="1173" height="374" alt="Image" src="https://github.com/user-attachments/assets/cfda2723-71f2-4695-bef1-259a91880ffe" />

Note that this also uses CISPO-style IS clipping, not IcePop.
